charts/core/src/Grid/Grid.tsx

```diff
@@ -15,10 +15,12 @@ const unsetGridOptions = {
 };
 
 export function Grid({ horizontal = true, vertical = true }: GridProps) {
-  const { updateChartOptions } = useChartContext();
+  const { chart } = useChartContext();
   const { theme } = useDarkMode();
 
   useEffect(() => {
+    if (!chart.ready) return;
```

Comment:
I might be misunderstanding how echarts works, but why can't we check if `chart` is defined?

Reply:
`chart` in this case is an instance of `ChartInstance`. In theory, that's actually always defined even when the `echart` hasn't been initialized. We could check if `chart._echartsInstance` is defined, but I just added `chart.ready` as a helper boolean to avoid having to interact directly with the instance outside of `useEchart`. Does that make sense?